smithy-rules-engine/src/main/java/software/amazon/smithy/rulesengine/validators/AuthSchemesValidator.java

```diff
@@ -139,6 +140,34 @@ private Stream<ValidationError> validateSigv4(
                     propertyIs(authScheme, SIGNING_REGION, Literal::asString));
         }
 
+        private Stream<ValidationError> hasAllKeys(
+            Map<Identifier, Literal> map,
+            List<Identifier> keys,
+            SourceLocation sourceLocation
+        ) {
+            return keys.stream().flatMap(key -> {
+                if (!map.containsKey(key)) {
+                    return Stream.of(new ValidationError(ValidationErrorType.INVALID_AUTH_SCHEMES,
+                        String.format("Unexpected key: `%s` (valid keys: %s)", key, keys), sourceLocation));
+                } else {
+                    return Stream.empty();
+                }
+            });
+        }
+
+        private Stream<ValidationError> validateBeta(
+            Map<Identifier, Literal> authScheme,
+            SourceLocation sourceLocation
+        ) {
+            List<ValidationError> missingKeys = hasAllKeys(authScheme,
+                ListUtils.of(SIGNING_NAME, SIGNING_REGION, NAME), sourceLocation)
+                .collect(Collectors.toList());
+            if (!missingKeys.isEmpty()) {
+                return missingKeys.stream();
+            }
+            return propertyIs(authScheme, SIGNING_NAME, Literal::asString);
```

Comment:
Does this need to validate region explicitly like the Sigv4 implementation?

Reply:
since, its beta for beta auth methods and those might not have strict region requirements i decided to skip it here